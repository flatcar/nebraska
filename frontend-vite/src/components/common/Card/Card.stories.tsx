import { Meta, StoryFn } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import {
  CardDescriptionLabel,
  CardFeatureLabel,
  CardHeader,
  CardHeaderProps,
  CardLabel,
} from './Card';

export default {
  title: 'Card',
} as Meta;

export const FeatureLabel = () => <CardFeatureLabel>Feature Label</CardFeatureLabel>;

export const DescriptionLabel = () => (
  <CardDescriptionLabel>Description Label</CardDescriptionLabel>
);

export const Label = () => <CardLabel>Label</CardLabel>;
export const LabelWithStyles = () => <CardLabel labelStyle={{}}>Label</CardLabel>;

const CardHeaderTemplate: StoryFn<CardHeaderProps> = args => (
  <MemoryRouter>
    <CardHeader {...args} />
  </MemoryRouter>
);

export const Header = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header description',
    cardMainLinkPath: '/apps/123',
    cardMainLinkLabel: 'Flatcar',
    children: 'CardHeader children',
    cardTrack: 'a cardTrack',
  },
};

export const HeaderWithDescription = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header description',
  },
};

export const HeaderWithLink = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header With Link description',
    cardMainLinkPath: '/apps/123',
    cardMainLinkLabel: 'Flatcar',
  },
};

export const HeaderWithChildren = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header With Children description',
    children: 'CardHeader children',
  },
};

export const HeaderWithCardTrack = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header With cardTrack description',
    cardTrack: 'a cardTrack',
  },
};

export const HeaderWithCardId = {
  render: CardHeaderTemplate,

  args: {
    cardDescription: 'Card Header With cardId description',
    cardId: 'a cardId',
  },
};
