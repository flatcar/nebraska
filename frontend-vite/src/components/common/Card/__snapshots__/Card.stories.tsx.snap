// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card DescriptionLabel smoke-test 1`] = `
<div class="MuiBox-root css-1yuhvjn">
  <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-14 css-1s23nr9-MuiTypography-root">
    Description Label
  </span>
</div>
`;

exports[`Card FeatureLabel smoke-test 1`] = `
<span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-featureLabel-4 css-1s23nr9-MuiTypography-root">
  Feature Label
</span>
`;

exports[`Card Header smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-37 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <h2 class="MuiTypography-root MuiTypography-h2 css-150084s-MuiTypography-root">
        <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover stable-makeStyles-mainLink-39 css-1emptwe-MuiTypography-root-MuiLink-root"
           href="/apps/123"
        >
          Flatcar
        </a>
      </h2>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-42 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
          a cardTrack
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-41 css-1s23nr9-MuiTypography-root">
          Card Header description
        </span>
      </div>
    </div>
  </div>
  <div class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root">
    CardHeader children
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-43 css-xetjfm-MuiDivider-root">
`;

exports[`Card HeaderWithCardId smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-87 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 stable-makeStyles-cardTitle-88 css-1s23nr9-MuiTypography-root">
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-92 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
          a cardId
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-91 css-1s23nr9-MuiTypography-root">
          Card Header With cardId description
        </span>
      </div>
    </div>
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-93 css-xetjfm-MuiDivider-root">
`;

exports[`Card HeaderWithCardTrack smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-77 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 stable-makeStyles-cardTitle-78 css-1s23nr9-MuiTypography-root">
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-82 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
          a cardTrack
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-81 css-1s23nr9-MuiTypography-root">
          Card Header With cardTrack description
        </span>
      </div>
    </div>
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-83 css-xetjfm-MuiDivider-root">
`;

exports[`Card HeaderWithChildren smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-67 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 stable-makeStyles-cardTitle-68 css-1s23nr9-MuiTypography-root">
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-72 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-71 css-1s23nr9-MuiTypography-root">
          Card Header With Children description
        </span>
      </div>
    </div>
  </div>
  <div class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root">
    CardHeader children
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-73 css-xetjfm-MuiDivider-root">
`;

exports[`Card HeaderWithDescription smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-47 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 stable-makeStyles-cardTitle-48 css-1s23nr9-MuiTypography-root">
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-52 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-51 css-1s23nr9-MuiTypography-root">
          Card Header description
        </span>
      </div>
    </div>
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-53 css-xetjfm-MuiDivider-root">
`;

exports[`Card HeaderWithLink smoke-test 1`] = `
<div class="MuiGrid-root MuiGrid-container stable-makeStyles-gridHeader-57 css-1xzqqmo-MuiGrid-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-j1w48a-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <h2 class="MuiTypography-root MuiTypography-h2 css-150084s-MuiTypography-root">
        <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover stable-makeStyles-mainLink-59 css-1emptwe-MuiTypography-root-MuiLink-root"
           href="/apps/123"
        >
          Flatcar
        </a>
      </h2>
    </div>
    <div class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root">
      <div class="MuiBox-root css-1yq8yj7">
        <p class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap stable-makeStyles-idLabel-62 css-1s94icn-MuiTypography-root"
           arial-label="group-id"
        >
        </p>
      </div>
      <div class="MuiBox-root css-1yuhvjn">
        <span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-descriptionLabel-61 css-1s23nr9-MuiTypography-root">
          Card Header With Link description
        </span>
      </div>
    </div>
  </div>
</div>
<hr class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light stable-makeStyles-innerDivider-63 css-xetjfm-MuiDivider-root">
`;

exports[`Card Label smoke-test 1`] = `
<span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-label-26 stable-makeStyles-label-27 css-1s23nr9-MuiTypography-root">
  Label
</span>
`;

exports[`Card LabelWithStyles smoke-test 1`] = `
<span class="MuiTypography-root MuiTypography-body1 stable-makeStyles-label-35 stable-makeStyles-label-36 css-1s23nr9-MuiTypography-root">
  Label
</span>
`;
